### PR TITLE
Fix thrust cut when landed

### PR DESCRIPTION
**Pull request by MaEtUgR**
**Describe problem solved by this pull request**
Land detector issues became more frequent and thanks a lot to @RomanBapst for finding the following problems:
1. cutting the thrust when landing is broken and reported
1. the attitude setpoint plots generated from the log are incomplete even though the topic is published

**Describe your solution**
1. I realized I did a mistake in https://github.com/PX4/Firmware/pull/13262/files#diff-fc77c3ef569029d45764664c75d4b0c1R688 and fix it in the first commit of this pr.
   Commit message:
   ```
   The problem was that before the input to the attitude setpoint generation
   was adjusted to generate a level attitude with zero thrust keeping the
   heading. I refactored the PositionControl class in #13262 to directly
   generate the attitude setpoint output. So here I'm adjusting the attitude
   setpoint to do the exact same thing as before but without interleaving
   with the PositionControl logic.
   ```
1. I realized (thanks to @bkueng's input) that I did a mistake in https://github.com/PX4/Firmware/pull/13262/files#diff-fc77c3ef569029d45764664c75d4b0c1R692-R693 and fix it in the second commit of this pr.
   Commit message:
   ```
   The plot of the attitude setpoint in the log did not show any values
   because the message timestamp that the position control module sets
   was overwritten by the PositionControl attitude generation.
   ```

**Test data / coverage**
SITL tested with very close attention to the logged values of thrust cutting and attitude setpoint when landing with an odd heading.

**Additional context**
FYI @nicovanduijn

### CODE REVIEW

**Comment by MaEtUgR** [comment]
I just fixed the interface also for testing. Issue found by CI.

File: src/modules/mc_pos_control/PositionControl/ControlMath.hpp [hunk 1]
```diff
@@ -51,7 +51,8 @@ namespace ControlMath
  * @param yaw_sp the desired yaw
  * @return vehicle_attitude_setpoints_s structure
  */
-vehicle_attitude_setpoint_s thrustToAttitude(const matrix::Vector3f &thr_sp, const float yaw_sp);
+void thrustToAttitude(vehicle_attitude_setpoint_s &attitude_setpoint, const matrix::Vector3f &thr_sp,
```
Comment (bkueng) [inline src/modules/mc_pos_control/PositionControl/ControlMath.hpp]: I'd turn it around: the convention is input params, then output params:
void thrustToAttitude(const matrix::Vector3f &thr_sp, const float yaw_sp, vehicle_attitude_setpoint_s &attitude_setpoint);

Comment (MaEtUgR) [reply src/modules/mc_pos_control/PositionControl/ControlMath.hpp]: I can totally do that I just didn't know where this convention is.
I was following the http://www.cplusplus.com/reference/cstring/memcpy/ convention which might not be applicable.

Comment (MaEtUgR) [reply src/modules/mc_pos_control/PositionControl/ControlMath.hpp]: Maybe this is worth taking an example from: https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rf-out-multi :+1:

Comment (MaEtUgR) [reply src/modules/mc_pos_control/PositionControl/ControlMath.hpp]: I'm convinced, fixed this in https://github.com/PX4/Firmware/pull/13381/commits/3d845dea9d31c4f4ab7d2989bb48f1e60698922d and will follow the convention from now on.

**Comment by RomanBapst** [comment]
@MaEtUgR Having a quick look now.

**Comment by RomanBapst** [comment]
@MaEtUgR The changes look good. Something I haven't quite figured out yet is what the thrust values in vehicle_local_position_setpoint should be. As far as I can see they are copied from the controller but before the limiting happens. Is this what we want?

**Comment by MaEtUgR** [comment]
@RomanBapst Yes exactly, that was already the case before. See https://github.com/PX4/Firmware/pull/13262/files#diff-fc77c3ef569029d45764664c75d4b0c1L684-L694

Note that I plan to get rid of the normalized thrust setpoint and replace it with acceleration: https://github.com/PX4/Firmware/pull/12072/files#diff-e09a66ab293d2aa0badec8b7f358bf6dL224
I'll do further incremental split out prs from that, the next one is ready to review here #13347 .

**Comment by julianoes** [comment]
@MaEtUgR can you check if this is also required for v1.10?

Closes #13388.

**Comment by MaEtUgR** [comment]
@julianoes It's not required or portable since 1.10 does not contain the bug. :heavy_check_mark:

**Pull request merged**